subsys/net/l2/ethernet/arp.c

```diff
@@ -492,8 +510,14 @@ static inline struct net_pkt *arp_prepare_reply(struct net_if *iface,
 	hdr->protolen = sizeof(struct in_addr);
 	hdr->opcode = htons(NET_ARP_REPLY);
 
-	memcpy(&hdr->dst_hwaddr.addr, &eth_query->src.addr,
-	       sizeof(struct net_eth_addr));
+	if (dst_addr) {
```

Comment:
Instead of this if here I would always memcpy() from dst_addr (and see below why)

Reply:
sure, makes sense